frontend/src/pages/list/BookList.tsx

```diff
@@ -1,121 +1,349 @@
+  pageCount: DataTableFilterMetaData;
+  width: DataTableFilterMetaData;
+  height: DataTableFilterMetaData;
+  thickness: DataTableFilterMetaData;
+  retailPrice: DataTableFilterMetaData;
+}
+
+export default function BookList() {
+  const emptyBook = {
+    id: 0,
+    title: "",
+    authors: [""],
+    genres: [""],
+    isbn13: "",
+    isbn10: "",
+    publisher: "",
+    publishedYear: 0,
+    pageCount: 0,
+    width: 0,
+    height: 0,
+    thickness: 0,
+    retailPrice: 0,
+  };
+
+  // Custom dropdown selector for Genre
+  const [selectedGenre, setSelectedGenre] = useState<string>("");
+
+  const genreFilter = () => {
+    return (
+      <Dropdown
+        value={selectedGenre}
+        options={GENRE_DATA.map((genreRow) => genreRow.genre)}
+        appendTo={"self"}
+        onChange={(e) => setSelectedGenre(e.value)}
+        placeholder={"Select Genre"}
+        showClear
+      />
+    );
+  };
+
+  // Properties of each column that change, the rest are set below when creating the actual Columns to be rendered
+  const COLUMNS: TableColumn[] = [
+    { field: "id", header: "ID", filterPlaceholder: "Search by ID" },
+    { field: "title", header: "Title", filterPlaceholder: "Search by Title" },
+    {
+      field: "authors",
+      header: "Authors",
+      filterPlaceholder: "Search by Authors",
+    },
+    {
+      field: "genres",
+      header: "Genre",
+      filterPlaceholder: "Search by Genre",
+      customFilter: genreFilter,
+    },
+    { field: "isbn13", header: "ISBN", filterPlaceholder: "Search by ISBN" },
+    {
+      field: "isbn10",
+      header: "ISBN",
+      filterPlaceholder: "Search by ISBN",
+      hidden: true,
+    },
+    {
+      field: "publisher",
+      header: "Publisher",
+      filterPlaceholder: "Search by Publisher",
+      hidden: true,
+    },
+    {
+      field: "publishedYear",
+      header: "Publication Year",
+      filterPlaceholder: "Search by Publication Year",
+      hidden: true,
+    },
+    {
+      field: "pageCount",
+      header: "Page Count",
+      filterPlaceholder: "Search by Page Count",
+      hidden: true,
+    },
+    {
+      field: "width",
+      header: "Width",
+      filterPlaceholder: "Search by Width",
+      hidden: true,
+    },
+    {
+      field: "height",
+      header: "Height",
+      filterPlaceholder: "Search by Height",
+      hidden: true,
+    },
+    {
+      field: "thickness",
+      header: "Thickness",
+      filterPlaceholder: "Search by Thickness",
+      hidden: true,
+    },
+    {
+      field: "retailPrice",
+      header: "Retail Price",
+      filterPlaceholder: "Search by Price",
+    },
+  ];
+
+  // State to track the current book that has been selected to be deleted
+  const [selectedDeleteBook, setSelectedDeleteBook] = useState<Book>(emptyBook);
+
+  // Custom body template for edit/delete buttons
+  const actionBodyTemplate = (rowData: Book) => {
+    return (
+      <React.Fragment>
+        <Button
```

Comment:
You have a util component called edit and delete that you don't use. Maybe you forgot to refactor or it isn't possible to be refactored?

Reply:
Actually the other pages were refactored for this. So you should try this one too.